File: CS50_CS/Problems/Week_2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    string text = get_string("Text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if ((text[i] >= 65 && text[i] <= 90) || (text[i] >= 97 && text[i] <= 122))
        {
            letters++;
        }
        else if (isspace(text[i]) != 0)
        {
            words++;
        }
        else if (text[i] == 33 || text[i] == 46 || text[i] == 63)
        {
            sentences++;
        }
    }
    float L = (letters * 100) / (float) words;
    float S = (sentences * 100) / (float) words;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
}

File: CS50_CS/Problems/Week_6/readability.py
def main():
    text=input("Text: ")
    letters=0
    words=1
    sentences=0
    for sign in text:
        if sign.isalpha()==True:
            letters+=1
        elif sign==" ":
            words+=1
        elif sign=="!" or sign=="." or sign=="?":
            sentences+=1
    L=(letters*100)/words
    S=(sentences*100)/words
    index=round(0.0588*L-0.296*S-15.8)
    if index<1:
        return print("Before Grade 1")
    elif index>16:
        print("Grade 16")
    else:
        print(f"Grade {index}")
    

if __name__=="__main__":
    main()